### PR TITLE
Add onX support

### PR DESCRIPTION
**Pull request by Siriah**
Hi!

There were a lot of false positive errors in my workspace from combining Lombok constructors with constructor injection, so I implemented onX support in your plugin. I hope it'll also be useful for everyone currently +1-ing Issue #47 :)

These commits will suppress the errors from using the onConstructor/onMethod/onParam parameters, and will also cause the annotations to be added to the generated methods when Delomboking.

### CODE REVIEW

**Comment by mplushnikov** [comment]
Thank you very much Siriah! I will look at it!

File: lombok-plugin/src/main/java/de/plushnikov/intellij/plugin/handler/OnXAnnotationHandler.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+    
+    PsiElement highlightedElement = file.findElementAt(highlightInfo.getStartOffset());
+    
+    PsiNameValuePair nameValuePair = findContainingNameValuePair(highlightedElement);
+    if (nameValuePair == null || !(nameValuePair.getContext() instanceof PsiAnnotationParameterList)) {
+      return false;
+    }
+
+    String parameterName = nameValuePair.getFirstChild().getText();
```
Comment (mplushnikov) [inline lombok-plugin/src/main/java/de/plushnikov/intellij/plugin/handler/OnXAnnotationHandler.java]: Could it be just: "nameValuePair.getName()"?

File: lombok-plugin/src/main/java/de/plushnikov/intellij/plugin/handler/OnXAnnotationHandler.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    if (!ONX_PARAMETERS.contains(parameterName)) {
+      return false;
+    }
+    
+    PsiElement containingAnnotation = nameValuePair.getContext().getPrevSibling().getContext();
```
Comment (mplushnikov) [inline lombok-plugin/src/main/java/de/plushnikov/intellij/plugin/handler/OnXAnnotationHandler.java]: Why not just "nameValuePair.getContext().getContext()"?

**Comment by mplushnikov** [comment]
Works really well, I will merge it next days!

**Comment by Siriah** [comment]
Awesome! I'll fix those needlessly convoluted bits of code you mentioned in the diff comments.

**Comment by danlangford** [comment]
as the owner of #47 me, and my inbox, thank you so so much.

**Comment by anitasv** [comment]
Cool. :+1:

**Comment by aquator** [comment]
Awesome, can't wait the new version. Thanks a lot!

**Comment by mogren** [comment]
Great, works for me so far! :-)

**Pull request merged**